build/Dockerfile

```diff
@@ -1,43 +1,44 @@
-# build
-FROM --platform=linux/amd64 node:16 AS build
+FROM --platform=linux/arm64 node:16-alpine AS deps
+RUN apk add --no-cache libc6-compat
+WORKDIR /app
 
-WORKDIR /opt/app
+COPY package.json yarn.lock ./
+RUN \
+  if [ -f yarn.lock ]; then yarn --frozen-lockfile; \
+  else echo "Lockfile not found." && exit 1; \
+  fi
 
-COPY package*.json ./
 
-RUN yarn install
+FROM --platform=linux/arm64 node:16-alpine AS builder
```

Comment:
we need to remove the platform and let the docker choose based on the architecture of the node where the runner is running